test/Microsoft.Azure.WebJobs.Host.FunctionalTests/JobHostMetadataProviderTests.cs

```diff
@@ -139,6 +200,22 @@ public void DefaultTypeForQueue()
             // so byte[] is the safest default. 
             var t3 = metadataProvider.GetDefaultType(new QueueAttribute("q"), FileAccess.Write, null);
             Assert.Equal(typeof(IAsyncCollector<byte[]>), t3);
-        }      
+        }
+
+        [Fact]
+        public void TestBatchFluentDefaultType()
+        {
+            IHost host = new HostBuilder()
+                .ConfigureDefaultTestHost(b =>
+                {
+                    b.AddExtension<FluentExtensionConfig>();
+                })
+                .Build();
+
+            var metadataProvider = host.CreateMetadataProvider();
+
+            var dataType = metadataProvider.GetDefaultType(new TestAttribute(), FileAccess.Read, typeof(object[]));
```

Comment:
Should also work for SomeData[] and string[] right? Can you also add test cases for that?

Reply:
Good point! I think SomeData[] would never really be requested (if I understand correctly), given this is only called by out of proc languages. string[] is definitely a valid case. Either way, I will add tests for both because they should work.